src/fabric.c

```diff
@@ -485,10 +498,45 @@ void ofi_create_filter(struct fi_filter *filter, const char *raw_filter)
 		++raw_filter;
 	}
 
-	filter->names= ofi_split_and_alloc(raw_filter, ",", NULL);
+	filter->names = ofi_split_and_alloc(raw_filter, ",", NULL);
 	if (!filter->names)
 		FI_WARN(&core_prov, FI_LOG_CORE,
 			"unable to parse filter from: %s\n", raw_filter);
+
+	tkn_str = strtok((char *)raw_filter, ";");
+	if (!tkn_str)
+		return;
+
+	while (tkn_str != NULL) {
+		count_prov_names++;
+		tkn_prove_names[i++] = tkn_str;
+		tkn_str = strtok (NULL, ";");
+	}
+
+	for(i = 0; i < count_prov_names; i++) {
+		if(!ofi_getprov(tkn_prove_names[i], strlen(tkn_prove_names[i]))) {
+			FI_WARN(&core_prov, FI_LOG_CORE,
+				"provider %s is unknown, misspelled or DL provider?\n",
+				tkn_prove_names[i]);
+			not_found_names++;
+		}
+	}
+
+	for (prov = prov_head; prov; prov = prov->next) {
+		for (i = 0; i < not_found_names; i++) {
+			if (strlen(prov->prov_name) != strlen(tkn_prove_names[i])
+				&& !strncasecmp(prov->prov_name, tkn_prove_names[i],
+						strlen(tkn_prove_names[i]))) {
+				if (strlen(tkn_prove_names[i]) > 5)
+					ofi_closest_prov_names(prov->prov_name,
+							       tkn_prove_names[i], 5);
+				else
+					ofi_closest_prov_names(prov->prov_name,
+							       tkn_prove_names[i], 2);
+			}
+		}
+	}
```

Comment:
Move the above code into the function mentioned above.  The code will need to be adjusted somewhat, since the unmatched string will already be given as input.

Reply:
@shefty  Done.

Reply:
Move the above logic into a new call.  You need to iterate over all the filter->names, since any of them could be able to be split.  Example:
```
static void verify_filter_names(char **names)
{
    for (i = 0; names[i]; i++) {
        split_names = ofi_split_and_alloc(names[i], ";", NULL)
        if (!split_names) {
            FI_WARN...
            return;
        }

        for (j = 0; split_names[j]; j++) {
            if (!ofi_getprov(split_names[j], strlen(split_names[j]))) {
                FI_WARN...
                ofi_suggest_prov_names(split_names[j])
            }
        }
        ofi_free_string_array(split_names);
    }
}
```
Call this function from this location.

Reply:
@shefty Done. There is some issue with the `osx`. It doesn't relate to the changes.

Reply:
@shefty Done.